client/me/memberships/main.jsx

```diff
@@ -31,29 +31,48 @@ import './main.scss';
  */
 import noMembershipsImage from 'assets/images/illustrations/no-memberships.svg';
 
+const MembershipListDate = ( { translate, subscription, moment } ) => {
+	let endDate = translate( 'Never Expires' );
+	let endDateFromNow = '-';
```

Comment:
Optional and non-blocking: to make this a bit more readable, getters might help a bit for the bits that change, and we can keep the markup together in MembershipItem:

```jsx
const getMembershipEndDate = ( endDate ) => {
    if ( ! endDate ) {
        return translate( 'Never Expires );
    }
    return moment( subscription.end_date ).format( 'll' );
}

const getEndDateFromNow = ( endDate ) => {
    if ( ! endDate ) { 
        return '-';
    }
    return translate( 'Renews %s', { args: moment( subscription.end_date ).fromNow() } ); 
}
```

Reply:
I prefer your way. It is more readable. Thanks for the suggestion.